src/runtime/ndarray.cc

```diff
@@ -327,7 +336,10 @@ std::shared_ptr<SharedMemory> NDArray::GetSharedMem() const {
   return this->data_->mem;
 }
 
-bool NDArray::IsDataPinned(DLTensor* tensor) {
+bool NDArray::IsContainerPinned(NDArray::Container* ptr) {
+  if (ptr->pinned_by_dgl_)
+    return true;
+  auto* tensor = &(ptr->dl_tensor);
```

Comment:
Since we have added a flag, when do we need to check the pointer again?

Reply:
If we borrowed a tensor pinned by PyTorch, we don't want to unpin it automatically. But we still need to know whether it's pinned.